servicetalk-grpc-internal/src/main/java/io/servicetalk/grpc/internal/StatusMessageUtils.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.grpc.internal;
+
+import io.servicetalk.http.api.HttpHeaders;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import javax.annotation.Nullable;
+
+import static io.servicetalk.buffer.api.CharSequences.newAsciiString;
+
+/**
+ * Provides utilities around percent-encoding and decoding the GRPC status message.
+ * <p>
+ * Note that much of the actual encoding and decoding logic is borrowed from the {@code io.grpc.Status} class,
+ * specifically the {@code io.grpc.Status#StatusMessageMarshaller}.
+ */
+public final class StatusMessageUtils {
```

Comment:
Just in case we will need to add more utilities around that status in the future, consider renaming it to `GrpcStatusUtils`